### PR TITLE
DeBank Extension

### PR DESCRIPTION
**Pull request by Yukaii**
## Description

New Extension built with [DeBank Open API](https://openapi.debank.com/docs)

## Type of change

- New extension

### Add to favorite

https://user-images.githubusercontent.com/4230968/147852307-b72e273e-a4da-46fb-a1db-85349a781da2.mp4

### Load from favorite

https://user-images.githubusercontent.com/4230968/147852211-c6c678c3-89c5-41ce-b608-abe1cf50e756.mp4

### Use address to search

https://user-images.githubusercontent.com/4230968/147852215-7d2b7705-b1e3-4ddc-820e-41b8cdd1f94d.mp4

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)

### CODE REVIEW

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @Yukaii 👋 

It looks nice, even though I don't have a way to fully test it I guess 😄 Could you remove the `command-icon.png` icon from assets, no need to have it there 😄 

When it's removed, I think we're ready to get it merged 💯 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Review by pernielsentikaer — APPROVED** [review]
Hi @Yukaii 👋 

Thank you for the existing extension, I'm sure it's possible to extend further later on 💪 

I'm going to accept it and then will my colleague do a final review before it's finally merged 🚀

File: extensions/debank/src/index.tsx [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+              />
+            )}
+          </ActionPanel>
+        )
+      }
+    />
+  );
+
+  return (
+    <List onSearchTextChange={onChange}>
```
Comment (mathieudutour) [inline extensions/debank/src/index.tsx:75]: What do you think about setting the `isLoading` prop when you are loading the address?

File: extensions/debank/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+      });
+    }
+  }, []);
+
+  const { push } = useNavigation();
+
+  if (!balance) {
+    return (
+      <List>
+        <List.Item key="1" title="Loading..." />
```
Comment (mathieudutour) [inline extensions/debank/src/index.tsx:171]: the recommended UX when loading a list content is to render an empty List with `isLoading` set to true (https://developers.raycast.com/api-reference/user-interface/list#list)

File: extensions/debank/src/index.tsx [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+      });
+    }
+  }, []);
+
+  const hasSomething = (protocols && protocols?.length > 0) || (tokens && tokens?.length > 0);
+
+  if (!protocols || !tokens) {
+    return (
+      <List>
+        <List.Item key="1" title="Loading..." />
```
Comment (mathieudutour) [inline extensions/debank/src/index.tsx:257]: same as above

**Review by mathieudutour — CHANGES_REQUESTED** [review]
Looking good, but the UX for loading Lists could be improved (see comments)

**Review by mathieudutour — APPROVED** [review]
Looks good, thanks!

**Pull request merged**